polytracker/src/passes/bb_splitting_pass.cpp

```diff
@@ -41,34 +81,10 @@ BBSplittingPass::analyzeBasicBlock(BasicBlock &basicBlock) const {
           continue;
         }
       }
-      std::string fname;
-      if (auto function = call->getCalledFunction()) {
-        if (function->hasName()) {
-          fname = function->getName().data();
-        }
-        // Note (Carson): Changed getCalledValue() --> Operand()
-        // https://reviews.llvm.org/D78882
-      } else if (auto v = call->getCalledOperand()->stripPointerCasts()) {
-        if (v->hasName()) {
-          fname = v->getName().data();
-        }
-      }
       // We need to split this BB into a new one after the call
       auto bb = next->getParent();
-      // Don't bother logging these common functions, but still split for them
-      bool includeFunctionName =
-          (fname != "llvm.dbg.declare" && fname != "llvm.dbg.value" &&
-           fname != "llvm.lifetime.end.p0i8" && fname != "__assert_fail");
 #ifdef DEBUG_INFO
-      if (fname.length() == 0 || bb->hasName() || includeFunctionName) {
-        if (bb->hasName()) {
-          llvm::errs() << " " << bb->getName().data();
-        }
-        if (includeFunctionName) {
-          llvm::errs() << " after call to " << fname;
-        }
-        llvm::errs() << "\n";
-      }
+      log_function(bb, get_function_name(call));
```

Comment:
Do you think you could move the conditionally compiled code out if `analyzeBasicBlock` and provide an empty `log_function` for release builds and this one for debug?

Reply:
What are the advantages of having it like that? I'm partial to this approach since having:
```c++
void log_function() { // implementation }

std::vector analyzeBasicBlock() {
// ... some code

#ifdef DEBUG_INFO
    log_function();
#endif

///... other code
}
```
makes it clear to readers of `analyzeBasicBlock` that `log_function` only matters for debug builds. 
If there were several places where `log_function` is called, then I would definitely agree to move the debug check there (to avoid duplication and noise). But since this is a helper function that is only used once, I think this way is a bit clearer.

What do you think?
```

Reply:
The advantage would be that we could later re-use it in other places in polytracker. But at that point it would probably be better just to use a proper logging library like [spdlog](https://github.com/gabime/spdlog).